.github/workflows/main.yml

```diff
@@ -0,0 +1,77 @@
+name: Generate new docs
+
+on:
+  workflow_dispatch:
+    inputs:
+      project:
+        description: 'ember or ember-data'     
+        required: true
+      version:
+        description: 'The version to build, i.e. 3.20.0'     
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Clone the project repo
+        run: |
+          set -euo pipefail
+          if [$INPUT_PROJECT -eq 'ember']
+          then
+            git clone https://github.com/emberjs/ember.js
+            cd ember.js
+            git checkout "v$INPUT_VERSION"
+          elif [$INPUT_PROJECT -eq 'ember-data']
+          then
+            git clone https://github.com/emberjs/data
+            cd data
+            git checkout "v$INPUT_VERSION"
+          else
+            echo "Please specify either ember or ember-data for the project."
+          fi
+
+      - name: Clone ember-api-docs-data
+        run: git clone https://github.com/ember-learn/ember-api-docs-data
+      
+      - name: yarn install
+        run: yarn install
+
+      - name: Generate s3-docs
+        working-directory: ember-jsonapi-docs
+        run: yarn gen --project $INPUT_PROJECT --version $INPUT_VERSION
+
+      - name: Generate json-docs
+        working-directory: ember-jsonapi-docs
+        run: yarn start --project $INPUT_PROJECT --version $INPUT_VERSION
+  
+  pull-request:
+    name: Open a pull request in ember-api-docs-data
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Git
+        run: |
+          git config --global user.name "Tomster"
+          git config --global user.email "tomster@emberjs.com"
+      - name: Set up SSH
+        uses: webfactory/ssh-agent@v0.1.1
+        with:
+          ssh-private-key: ${{ secrets.GUIDES_SOURCE_DEPLOY_KEY }}
+
+      - name: commit the changes
+        run: cd .. && git add . && git commit -m "${INPUT_PROJECT}-${INPUT_VERSION}"
+
+      - name: Push
+        working-directory: ember-api-docs-data
+        run: |
+          set -euo pipefail
+          IFS=$'\n\t'
+          if git diff --exit-code HEAD~; then
+            echo "Nothing to push"
+          else
+            git push -u origin "${INPUT_PROJECT}-${INPUT_VERSION}"
+          fi
```

Comment:
- name: Push changes
         working-directory: ember-api-docs-data
         run: |
           if git diff --exit-code HEAD~; then
             echo "Nothing to push"
           else
             git push origin HEAD:"${{ github.event.inputs.project }}-{{ github.event.inputs.version }}"
           fi

Reply:
You could set up a step here to automatically open the pull request, but you'll need a token to do it (since the default token for this job only have access to this repo), so it may be easier to setup another automation on the other repo that automatically open a pull request when a branch named `ember-*` or `ember-data-*` is pushed, which is what I did for the tutorial.

Reply:
In the long run, it may also be easier to just merge the two repos, by moving the scripts here into a `build-script` branch of the `ember-api-docs-data` repo. Once this is working, the main "user interface" is the "trigger workflow" dialog, and the node scripts in here are essentially just an implementation detail for the automation – they could in theory be reimplemented inline in this yaml file using other actions and some bash. Not that we have any reason to do reimplement it that way, but once it is working, very few people will need to know anything about the scripts or run them locally, so tucking it away into a branch on the other repo seems appropriate. Mainly it simplifies some of the access control, both by having one less repo to worry about, but also that actions running on the same repo has more automatic access to things on the same repo, etc. Anyway, that's a bit premature at this point.